src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -295,7 +296,7 @@ TaskActivity INameVersionObjectManager<TaskActivity>.GetObject(string name, stri
                 throw new InvalidOperationException(message);
             }
 
-            return new TaskActivityShim(this, executor, name);
+            return new TaskActivityShim(this, info.Executor, name);

```

Comment:
When we return this value, we are losing information about whether the Task is registered or not. I'm assuming this is intentional, but I think I need more context to understand the ramifications of this.